src/traces/scattermapbox/defaults.js

```diff
@@ -0,0 +1,86 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Lib = require('../../lib');
+
+var subTypes = require('../scatter/subtypes');
+var handleMarkerDefaults = require('../scatter/marker_defaults');
+var handleLineDefaults = require('../scatter/line_defaults');
+var handleTextDefaults = require('../scatter/text_defaults');
+var handleFillColorDefaults = require('../scatter/fillcolor_defaults');
+
+var attributes = require('./attributes');
+var scatterAttrs = require('../scatter/attributes');
+
+
+module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout) {
+    function coerce(attr, dflt) {
+        return Lib.coerce(traceIn, traceOut, attributes, attr, dflt);
+    }
+
+    function coerceMarker(attr, dflt) {
+        var attrs = (attr.indexOf('.line') === -1) ? attributes : scatterAttrs;
+
+        // use 'scatter' attributes for 'marker.line.' attr,
+        // so that we can reuse the scatter marker defaults
+
+        return Lib.coerce(traceIn, traceOut, attrs, attr, dflt);
+    }
+
+    var len = handleLonLatDefaults(traceIn, traceOut, coerce);
+    if(!len) {
+        traceOut.visible = false;
+        return;
+    }
+
+    coerce('text');
+    coerce('mode');
+
+    if(subTypes.hasLines(traceOut)) {
+        handleLineDefaults(traceIn, traceOut, defaultColor, coerce);
+        coerce('connectgaps');
+    }
+
+    if(subTypes.hasMarkers(traceOut)) {
+        handleMarkerDefaults(traceIn, traceOut, defaultColor, layout, coerceMarker);
+
+        // array marker.size and marker.color are only supported with circles
+
+        var marker = traceOut.marker;
+
+        if(marker.symbol !== 'circle') {
```

Comment:
No **per-pt** coloring on other shapes. Mapbox doesn't support that at the moment, but they are planning ([ref](https://www.mapbox.com/blog/data-driven-styling/)) on implementing it soon.

So, I didn't feel the need to hack in a solution (which would have involved adding one layer per pt) for this case.